playback/src/decoder/symphonia_decoder.rs

```diff
@@ -133,30 +132,34 @@ impl SymphoniaDecoder {
     }
 
     fn ts_to_ms(&self, ts: u64) -> u32 {
-        let time_base = self.decoder.codec_params().time_base;
-        let seeked_to_ms = match time_base {
+        match self.decoder.codec_params().time_base {
             Some(time_base) => {
-                let time = time_base.calc_time(ts);
-                (time.seconds as f64 + time.frac) * 1000.
+                let time = Duration::from(time_base.calc_time(ts));
+                time.as_millis() as u32
             }
             // Fallback in the unexpected case that the format has no base time set.
-            None => ts as f64 * PAGES_PER_MS,
-        };
-        seeked_to_ms as u32
+            None => (ts as f64 * PAGES_PER_MS) as u32,
+        }
     }
 }
 
 impl AudioDecoder for SymphoniaDecoder {
     fn seek(&mut self, position_ms: u32) -> Result<u32, DecoderError> {
-        let seconds = position_ms as u64 / 1000;
-        let frac = (position_ms as f64 % 1000.) / 1000.;
-        let time = Time::new(seconds, frac);
+        // "Saturate" the position_ms to the duration of the track if it exceeds it.
+        let mut target = Duration::from_millis(position_ms as u64);
+        let codec_params = self.decoder.codec_params();
+        if let (Some(time_base), Some(n_frames)) = (codec_params.time_base, codec_params.n_frames) {
+            let duration = Duration::from(time_base.calc_time(n_frames));
+            if target > duration {
+                target = duration;
```

Comment:
I was wondering if a seek to the beginning would make more sense if we receive a value out of bounds? But then again, the actually seek problem is probably related to the websocket handling, so it should probably be fixed there.

Reply:
I don't think so because:
1. saturating seeks is idiomatic audio behavior (pull the slider in an UI beyond a known end point, you'd expect it to stick to the end)
2. saturating will effectively make it skip to the next track, which is likely more robust behavior than rewinding the beginning of the current track, where the controller may repeat the invalid seek beyond end, effectively getting stuck in a loop

Reply:
Sure, sounds reasonable